draft-ietf-ppm-dap.md

```diff
@@ -1478,30 +1487,35 @@ processed further and MUST be removed from the candidate set.
 If the Helper's `PrepareStep` is of type `continued`, then the Leader proceeds
 as follows.
 
-Let `leader_outbound` denote the leader's prepare message and `helper_outbound`
-denote the helper's. The leader computes the next state transition as follows:
+Let `leader_prep_share` denote the Leader's prepare message-share and
+`helper_prep_share` denote the Helper's. The Leader computes the next state
+transition as follows:
 
 ~~~
-inbound = VDAF.prep_shares_to_prep(agg_param, [leader_outbound, helper_outbound])
-out = VDAF.prep_next(prep_state, inbound)
+prep_msg = VDAF.prep_shares_to_prep(agg_param, [
+    leader_prep_share,
+    helper_prep_share,
+])
+out = VDAF.prep_next(prep_state, prep_msg)
 ~~~
 
-where `[leader_outbound, helper_outbound]` is a vector of two elements. If
-either of these operations fails, then the leader marks the report as invalid.
-Otherwise it interprets `out` as follows: If this is the last round of the VDAF,
-then `out` is the aggregator's output share, in which case the aggregator
-finishes and stores its output share for further processing as described in
-{{collect-flow}}. Otherwise, `out` is the pair `(new_state, prep_msg)`, where
-`new_state` is its updated state and `prep_msg` is its next VDAF message (which
-will be `leader_outbound` in the next round of continuation). For the latter
-case, the helper sets `prep_state` to `new_state`.
+where `[leader_prep_share, helper_prep_share]` is a vector of two elements. If
+either of these operations fails, then the leader marks the report as invalid
+with error "prep_share_failed". Otherwise it interprets `out` as follows: If
+this is the last round of the VDAF, then `out` is the Leader's output share, in
+which case it stores the output share for further processing as described in
+{{collect-flow}}. Otherwise, `out` is the pair `(next_prep_state,
```

Comment:
I don't like the "interprets `out` as follows" phrasing, and I deliberately changed it the last time I was in this paragraph. The reason is that the text you have suggests that the leader is supposed to check what round it currently is ("[I]f this is the last round of the VDAF") and then interpret `out` accordingly. That's backwards: the way that the Leader knows it's on the last round is if `VDAF.prep_next` returns an output share. It's not up to DAP to decide if we're on the last round of the VDAF or if `out` is an output share. VDAF makes that decision.

Reply:
Hmm. can you take another look? I haven't changes this phrasing AFAICT. In particular the following pattern is in both paragraphs:
> If either of these operations fails, then ... . Otherwise, it interprets `out` as follows: If this is the last round of the VDAF, ... . Otherwise, `out` is the pair ...

Alternatively, do you have a suggestion of how you think this should be worded?

Reply:
Sorry, I think I got my wires crossed while reviewing this. You did indeed not change the wording here. I may or may not have a point about the interpretation thing, but we don't have to address that in this change.